### PR TITLE
Fixed #32860 -- Made docs permalinks focusable to improve accessibility.

### PR DESCRIPTION
**Pull request by tim-mccurrach**
Made permalinks focusable for better keyboard navigation.

See https://code.djangoproject.com/ticket/32860
See also: https://github.com/django/djangoproject.com/pull/1100

### CODE REVIEW

File: docs/_theme/djangodocs/static/djangodocs.css [hunk 1]
```diff
@@ -125,8 +125,9 @@ div.versionadded span.title, div.versionchanged span.title, span.versionmodified
 div.versionadded, div.versionchanged, div.deprecated { color:#555; }
 
 /*** p-links ***/
-a.headerlink { color: #c60f0f; font-size: 0.8em; padding: 0 4px 0 4px; text-decoration: none; visibility: hidden; }
-h1:hover > a.headerlink, h2:hover > a.headerlink, h3:hover > a.headerlink, h4:hover > a.headerlink, h5:hover > a.headerlink, h6:hover > a.headerlink, dt:hover > a.headerlink { visibility: visible; }
+a.headerlink { color: #c60f0f; font-size: 0.8em; padding: 0 4px 0 4px; text-decoration: none; opacity: 0; }
```
Comment (pope1ni) [inline docs/_theme/djangodocs/static/djangodocs.css:128]: Please maintain alphabetical ordering of properties:

```suggestion
a.headerlink { color: #c60f0f; font-size: 0.8em; opacity: 0; padding: 0 4px 0 4px; text-decoration: none; }
```

Please can you also change `padding: 0 4px 0 4px;` on `a.headerlink` to `margin-left: 4px;` to remove the spacing inside the focus outline?

```suggestion
a.headerlink { color: #c60f0f; font-size: 0.8em; margin-left: 4px; opacity: 0; text-decoration: none; }
```

**Review by pope1ni — CHANGES_REQUESTED** [review]
I tested this out by applying the style in developer tools - it works well. I've made a few suggestions below.

I did also notice that we get some link underline under the pilcrow for `<h1>` elements, e.g. the main "Django documentation" header. It isn't a problem for `<h2>`, `<h3>`, and `<h4>`. (Didn't look at `<h5>` or `<h6>`.) This happens when hovering over the `<h1>` but not the `<a>` inside and is due to the following rules:

```css
#bd a:link, #bd a:visited { text-decoration: underline; }
h2 a, h3 a, h4 a { text-decoration: none !important; }
```

I wonder if it is worth updating the sphinx theme to be styled like docs.djangoproject.com at some point... What do you think, @carltongibson?

**Comment by tim-mccurrach** [comment]
@pope1ni Thanks for the review. 

Nice catch re. the underline, that was very eagle-eyed. I've left it as it is for the time being. I suspect `h1 a` could just be added to the rule you quoted above. But I don't know the docs amazingly well and don't want to inadvertently remove an underline elsewhere in the docs. Especially since if this is going to happen, it will solve it anyway:

> I wonder if it is worth updating the sphinx theme to be styled like docs.djangoproject.com at some point...

**Comment by felixxm** [comment]
@tim-mccurrach Thanks :+1:

**Pull request merged**